src/pool/srv_target.c

```diff
@@ -1112,15 +1112,25 @@ ds_pool_start(uuid_t uuid)
 	return rc;
 }
 
+static void ds_pool_hdl_get(struct ds_pool_hdl *hdl);
 static void pool_tgt_disconnect(struct ds_pool_hdl *hdl);
 
 static void
 pool_tgt_disconnect_all(struct ds_pool *pool)
 {
-	struct ds_pool_hdl *hdl;
+	while (!d_list_empty(&pool->sp_hdls)) {
+		struct ds_pool_hdl *hdl;
 
-	while ((hdl = d_list_pop_entry(&pool->sp_hdls, struct ds_pool_hdl, sph_pool_entry)) != NULL)
+		/*
+		 * The handle will not be freed before we get our reference,
```

Comment:
Minor nit, I presume the segv is due to double removal from this list fixed on line 1121.  It might be useful to change this to 

"will not be _removed from the list and_ freed"

Reply:
@jolivier23, removing a d_list_t multiple times via `d_list_del_init` is okay (but doesn't read well, that's why I changed the `d_list_pop_entry`). It was the removal from the hash table causing the entry to be freed and then accessed by `d_list_del_init` that led to the segfault.